### PR TITLE
set ignored scope widely

### PR DESCRIPTION
**Pull request by sooahleex**
### Changes
set ignored_scope as op_addresses_to_skip too

### Reason for changes
pruning consistently tried to apply to ignored scope

### CODE REVIEW

File: nncf/common/pruning/model_analysis.py [hunk 1]
```diff
@@ -167,6 +167,9 @@ def propagate_can_prune_attr_up(self):
         for node in reversed_sorted_nodes:
             # Check all output nodes accept_pruned_input attribute
             out_nodes = self.graph.get_next_nodes(node)
+            if out_nodes == []:
+                self.can_prune[node.node_id] =False
```
Comment (vinnamkim) [inline nncf/common/pruning/model_analysis.py:171]: Could you please auto-format with pylint?
It seems that this package uses https://github.com/openvinotoolkit/nncf/blob/develop/.pylintrc.
Not only for this line `=False`.

**Comment by daniil-lyakhov** [comment]
Guys, looks like we were removed from nncf developers group. Please add me as an reviewer and don't merge this without our review!

**Comment by vinnamkim** [comment]
> Guys, looks like we were removed from nncf developers group. Please add me as an reviewer and don't merge this without our review!
> 
> cc: @negved

Please request to join at this link: https://github.com/orgs/openvinotoolkit/teams/nncf_pytorch-maintainers/members

**Comment by vinnamkim** [comment]
> > Guys, looks like we were removed from nncf developers group. Please add me as an reviewer and don't merge this without our review!
> > cc: @negved
> 
> Please request to join at this link: https://github.com/orgs/openvinotoolkit/teams/nncf_pytorch-maintainers/members

@negvet

**Comment by daniil-lyakhov** [comment]
@sooahleex, we already have mechanism for ignoring layers, it's implemented here: https://github.com/openvinotoolkit/nncf/blob/160990cba0473452955c494b2087c4f4160241c7/nncf/common/pruning/node_selector.py#L279-L305

it used at the last step of node selection here: https://github.com/openvinotoolkit/nncf/blob/160990cba0473452955c494b2087c4f4160241c7/nncf/common/pruning/node_selector.py#L218

It's essential to generate masks for each target node even if it's in a ignored scope because of the algorithm implementation. Ignoring layers during mask propagation could lead to inconsistent pruning layout!

File: nncf/common/pruning/operations.py [hunk 2]
```diff
@@ -204,6 +204,10 @@ def generate_output_mask(cls, node: NNCFNode, graph: NNCFGraph,
         if not not_empty_masks:
             return None
 
+        # if node is leaf node return None
+        if not graph.get_output_edges(node):
+            return None
+
```
Comment (daniil-lyakhov) [inline nncf/common/pruning/operations.py:210]: Looks like problem is more fundamental: some of the input node has non empty pruning masks. The result of such return could lead to inconsistent pruning layout. @sooahleex, let's try to inspect this graph together

**Comment by sooahleex** [comment]
> 



> @sooahleex, we already have mechanism for ignoring layers, it's implemented here:
> 
> https://github.com/openvinotoolkit/nncf/blob/160990cba0473452955c494b2087c4f4160241c7/nncf/common/pruning/node_selector.py#L279-L305
> 
> it used at the last step of node selection here:
> 
> https://github.com/openvinotoolkit/nncf/blob/160990cba0473452955c494b2087c4f4160241c7/nncf/common/pruning/node_selector.py#L218
> 
> It's essential to generate masks for each target node even if it's in a ignored scope because of the algorithm implementation. Ignoring layers during mask propagation could lead to inconsistent pruning layout!

I knew that part and I checked that works out for mask propagation. When I tried to do pruning, the error is occurred in `count_flops_and_weights_per_node` for the layer which I added in ignored scopes. For the function `count_flops_and_weights_per_node`, There is no part to check whether it is prunable. So I added `op_addresses_to_skip` to check whether the layer is prunable or not.

**Pull request closed without merging**